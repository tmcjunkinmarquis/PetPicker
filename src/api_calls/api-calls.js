const fetchUsers = ()=>{
  const url = 'locslhost: 3000 / api / v1 / users'

  console.log('I will fetch users')
}

const fetchPets = ()=>{
  const url = 'https://pet-picker-api.herokuapp.com/api/v1/users/1/pets'
  
  return fetch(url)
  .then((response)=>{
    return response.json()
  })
  .then((pets)=>{
    return pets
  })
  
}

export { fetchUsers, fetchPets };
