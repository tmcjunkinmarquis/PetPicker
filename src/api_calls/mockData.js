export const mockUserData = {
  id: 2, 
  name: "Theresa", 
  species_to_adopt: "dog", 
  search_radius: 5, lat: 0.333e2, 
  lng: 0.444e2, 
  role: "adopter", 
  created_at: "2018-09-18 00:43:08", 
  updated_at: "2018-09-18 00:43:08", 
  password_digest: "$2a$10$5Zm81iRt83CJ.Z4cp79BL.esDeYk8mLpsTzc3BhOA72...", 
  pic: "https://avatars2.githubusercontent.com/u/25600007?...", 
  description: "I absolutely love pups!"
};

export const mockPetData = {
  "id": 1,
  "name": "Bubba",
  "description": "I am a cute pupper",
  "pic": "http://www.bubba.com"
};

export const mockWelcomePetData = {
  id: 1, 
  name: "Wanda", 
  species: 0, 
  description: "Doggo ipsum heckin angery woofer wow such tempt fl...", 
  pic: "https://www.cesarsway.com/sites/newcesarsway/files...",  
  user_id: 4
};

export const mockMatchData = [
  {
    "id": 51,
    "name": "Mack",
    "description": "Poops a lot",
    "pic": "https://cdn2-www.dogtime.com/assets/uploads/2011/03/puppy-development-460x306.jpg"
  },
  {
    "id": 52, 
    "name":  "Felton", 
    "description": "Poops a lot", 
    "pic": "https://cdn2-www.dogtime.com/assets/uploads/2011/03/puppy-development-460x306.jpg"
  },
  { "id": 53, 
    "name": "Tana", 
    "description": "Poops a lot", 
    "pic": "https://cdn2-www.dogtime.com/assets/uploads/2011/03/puppy-development-460x306.jpg" }
];

