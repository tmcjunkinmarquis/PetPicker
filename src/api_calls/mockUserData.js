export const mockUserData = {
  "id": 1,
  "username": "Bubba",
  "password": "123",
  "description": "I am Bubba."
};