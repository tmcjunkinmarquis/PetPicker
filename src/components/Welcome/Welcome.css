.welcome {
  background-image: url("http://floridapolitics.com/wp-content/uploads/2017/11/Puppies.jpg");
  width: 100%;
  height: 100vh;
}

.nope-or-like {
width: 100%;
}
