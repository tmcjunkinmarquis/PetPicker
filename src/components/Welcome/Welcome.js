import React, { Component } from 'react';
import HaveWantPetButton from '../HaveWantPetButton/HaveWantPetButton';
import { fetchWelcomePet } from '../../api_calls/api-calls';
import PetImage from '../PetImage/PetImage';
import './Welcome.css';

class Welcome extends Component {
  constructor(props) {
    super(props);
    this.state = {
      pet: {},
      loggedIn: false
    };
  }

  loadWelcomePet = async () => {
    const pet = await fetchWelcomePet();
    this.setState({ pet });
  }

  makePet = (pet) => {
    return (
      <PetImage pet={pet} />
    );
  }

  async componentDidMount() {
    await this.loadWelcomePet();
  }

  render() {
    return (
      <div>
        <h1>Welcome to Pet Picker!</h1>

        <div>{this.makePet(this.state.pet)}</div>

        <div className="nope-or-like">
          <HaveWantPetButton />
        </div>
      </div>
    );
  }
}

export default Welcome;
