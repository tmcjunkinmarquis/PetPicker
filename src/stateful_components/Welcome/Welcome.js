import React, { Component } from 'react';
import NopeButton from '../../stateless_components/NopeButton/NopeButton';
import LikeButton from '../../stateless_components/LikeButton/LikeButton';
import { fetchUsers, fetchPets } from '../../api_calls/api-calls';
import PetsContainer from '../../stateless_components/PetsContainer/PetsContainer';
import PetDescription from '../../stateless_components/PetDescription/PetDescription';
import './Welcome.css';

class Welcome extends Component {
  constructor(props) {
    super(props);
    this.state = {
      petsArray:[],
      loggedIn: true
    };
  }

  handleNopeClick = () => {
    console.log('howdy');
    
  }

  loadAllPets =  async ()=>{
    const petsArray = await fetchPets();
    this.setState({petsArray}); 
    
  }
    
  async componentDidMount(){
    await this.loadAllPets();
  }

  render() {
    return (
      <div>
        <h1>Welcome to Pet Picker!</h1>

        <PetsContainer 
          petsArray={this.state.petsArray}
          loggedIn={this.state.loggedIn}
        />
        
        <div className="nope-or-like">
          <NopeButton handleNopeClick={this.handleNopeClick} />
          <LikeButton handleLikeClick={this.handleLikeClick} />
        </div>
        
        <PetDescription />
      </div>
    );
  }
}

export default Welcome;