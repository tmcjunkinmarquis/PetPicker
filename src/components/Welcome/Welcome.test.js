/* eslint-disable max-len */

import React from 'react';
import { shallow } from 'enzyme';
import Welcome from './Welcome';
import { fetchWelcomePet } from '../../api_calls/api-calls';
jest.mock('../../api_calls/api-calls.js', () => {
  return {
    fetchWelcomePet: jest.fn()
  };
});

describe('welcome', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {};
    wrapper = shallow(<Welcome {...mockProps} />, {
      disableLifecycleMethods: true
    });
    jest.resetAllMocks();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('loadWelcomePet', () => {
    it('calls fetchWelcomePet', async () => {
      await wrapper.instance().loadWelcomePet();

      expect(fetchWelcomePet).toHaveBeenCalledTimes(1);
    });

    it('sets state.pet with pet ', async () => {
      const expected = {
        name: 'betsy'
      };

      console.log('fetchWelcomePet: ', fetchWelcomePet);

      fetchWelcomePet.mockImpementation(() => {
        return { name: 'betsy' };
      });

      await wrapper.instance().loadWelcomePet();

      expect(wrapper.state().pet).toEqual(expected);
    });
  });
});
