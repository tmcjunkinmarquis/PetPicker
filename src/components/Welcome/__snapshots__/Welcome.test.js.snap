// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`welcome matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="nope-or-like"
>
        <withRouter(HaveWantPetButton) />
</div>,
      "className": "welcome valign-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <withRouter(HaveWantPetButton) />,
        "className": "nope-or-like",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nope-or-like"
>
          <withRouter(HaveWantPetButton) />
</div>,
        "className": "welcome valign-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <withRouter(HaveWantPetButton) />,
          "className": "nope-or-like",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
